huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_directory.go

```diff
@@ -0,0 +1,332 @@
+			d.Set("root_id", utils.PathSearch("root_id", directory, nil)),
+			d.Set("qualified_name", utils.PathSearch("qualified_name", directory, nil)),
+			d.Set("created_at", utils.PathSearch("create_time", directory, nil)),
+			d.Set("updated_at", utils.PathSearch("update_time", directory, nil)),
+			d.Set("created_by", utils.PathSearch("create_by", directory, nil)),
+			d.Set("updated_by", utils.PathSearch("update_by", directory, nil)),
+			d.Set("children", utils.PathSearch(`children[*].name`, directory, make([]interface{}, 0))),
+		)
+		return diag.FromErr(mErr.ErrorOrNil())
+	}
+
+	return nil
+}
+
+func resourceStudioDirectoryUpdate(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	workspaceID := d.Get("workspace_id").(string)
+	dirType := d.Get("type").(string)
+
+	//nolint:misspell
+	updateDirectoryHttpUrl := "v2/{project_id}/design/directorys"
+	updateDirectoryProduct := "dataartsv2"
+
+	updateDirectoryClient, err := cfg.NewServiceClient(updateDirectoryProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio V2 Client: %s", err)
+	}
+	updateDirectoryPath := updateDirectoryClient.Endpoint + updateDirectoryHttpUrl
+	updateDirectoryPath = strings.ReplaceAll(updateDirectoryPath, "{project_id}", updateDirectoryClient.ProjectID)
+
+	updateDirectoryOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": workspaceID},
+	}
+
+	updateDirectoryOpt.JSONBody = utils.RemoveNil(buildUpdateDirectoryBodyParams(d, d.Get("directory_id").(string)))
+	updateDirectoryResp, err := updateDirectoryClient.Request("PUT", updateDirectoryPath, &updateDirectoryOpt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	updateDirectoryRespBody, err := utils.FlattenResponse(updateDirectoryResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	directory := utils.PathSearch("data.value", updateDirectoryRespBody, nil)
+
+	id, err := jmespath.Search("id", directory)
+	if err != nil {
+		return diag.Errorf("error updating DataArts Studio directory: %s is not found in API response", id)
+	}
+
+	qualifiedName, err := jmespath.Search("qualified_name", directory)
+	if err != nil {
+		return diag.Errorf("error updating DataArts Studio directory: %s is not found in API response", qualifiedName)
+	}
+	if qualifiedName == nil {
+		qualifiedName = d.Get("qualified_name")
+	}
+
+	rootID, err := jmespath.Search("root_id", directory)
+	if err != nil {
+		return diag.Errorf("error updating DataArts Studio directory: %s is not found in API response", rootID)
+	}
+	d.Set("root_id", rootID)
+
+	d.SetId(workspaceID + "/" + dirType + "/" + qualifiedName.(string))
+	d.Set("qualified_name", qualifiedName)
+
+	return nil
+}
+func buildUpdateDirectoryBodyParams(d *schema.ResourceData, id string) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"id":          id,
```

Comment:
the second param can be deleted?
```suggestion
		"id":          d.Get("directory_id").(string),
```